[2025-05-02T02:06:22.119+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-05-02T02:06:22.174+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: medical_dag_v17.dim_group.creating_dim_allergies manual__2025-05-02T02:06:20.068197+00:00 [queued]>
[2025-05-02T02:06:22.201+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: medical_dag_v17.dim_group.creating_dim_allergies manual__2025-05-02T02:06:20.068197+00:00 [queued]>
[2025-05-02T02:06:22.202+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-05-02T02:06:22.245+0000] {taskinstance.py:2890} INFO - Executing <Task(SparkSubmitOperator): dim_group.creating_dim_allergies> on 2025-05-02 02:06:20.068197+00:00
[2025-05-02T02:06:22.256+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'medical_dag_v17', 'dim_group.creating_dim_allergies', 'manual__2025-05-02T02:06:20.068197+00:00', '--job-id', '167', '--raw', '--subdir', 'DAGS_FOLDER/raw_enriched_dag.py', '--cfg-path', '/tmp/tmpce4ytv63']
[2025-05-02T02:06:22.259+0000] {standard_task_runner.py:105} INFO - Job 167: Subtask dim_group.creating_dim_allergies
[2025-05-02T02:06:22.263+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=5526) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-05-02T02:06:22.265+0000] {standard_task_runner.py:72} INFO - Started process 5527 to run task
[2025-05-02T02:06:22.327+0000] {task_command.py:467} INFO - Running <TaskInstance: medical_dag_v17.dim_group.creating_dim_allergies manual__2025-05-02T02:06:20.068197+00:00 [running]> on host airflow-scheduler
[2025-05-02T02:06:22.462+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='medical_dag_v17' AIRFLOW_CTX_TASK_ID='dim_group.creating_dim_allergies' AIRFLOW_CTX_EXECUTION_DATE='2025-05-02T02:06:20.068197+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-02T02:06:20.068197+00:00'
[2025-05-02T02:06:22.463+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-05-02T02:06:22.500+0000] {base.py:84} INFO - Retrieving connection 'spark_conn'
[2025-05-02T02:06:22.502+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master local --packages io.delta:delta-spark_2.12:3.3.0,org.apache.hadoop:hadoop-aws:3.3.4 --name arrow-spark --deploy-mode client /opt/etl/enriched_curated/creating_dim_allergies.py
[2025-05-02T02:06:24.855+0000] {spark_submit.py:649} INFO - :: loading settings :: url = jar:file:/opt/spark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2025-05-02T02:06:24.941+0000] {spark_submit.py:649} INFO - Ivy Default Cache set to: /home/***/.ivy2/cache
[2025-05-02T02:06:24.943+0000] {spark_submit.py:649} INFO - The jars for the packages stored in: /home/***/.ivy2/jars
[2025-05-02T02:06:24.947+0000] {spark_submit.py:649} INFO - io.delta#delta-spark_2.12 added as a dependency
[2025-05-02T02:06:24.948+0000] {spark_submit.py:649} INFO - org.apache.hadoop#hadoop-aws added as a dependency
[2025-05-02T02:06:24.949+0000] {spark_submit.py:649} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-4ac394d8-c752-48c7-ab53-b7934fa07484;1.0
[2025-05-02T02:06:24.950+0000] {spark_submit.py:649} INFO - confs: [default]
[2025-05-02T02:06:25.119+0000] {spark_submit.py:649} INFO - found io.delta#delta-spark_2.12;3.3.0 in central
[2025-05-02T02:06:25.166+0000] {spark_submit.py:649} INFO - found io.delta#delta-storage;3.3.0 in central
[2025-05-02T02:06:25.195+0000] {spark_submit.py:649} INFO - found org.antlr#antlr4-runtime;4.9.3 in central
[2025-05-02T02:06:25.276+0000] {spark_submit.py:649} INFO - found org.apache.hadoop#hadoop-aws;3.3.4 in central
[2025-05-02T02:06:25.297+0000] {spark_submit.py:649} INFO - found com.amazonaws#aws-java-sdk-bundle;1.12.262 in central
[2025-05-02T02:06:25.323+0000] {spark_submit.py:649} INFO - found org.wildfly.openssl#wildfly-openssl;1.0.7.Final in central
[2025-05-02T02:06:25.357+0000] {spark_submit.py:649} INFO - :: resolution report :: resolve 391ms :: artifacts dl 18ms
[2025-05-02T02:06:25.358+0000] {spark_submit.py:649} INFO - :: modules in use:
[2025-05-02T02:06:25.359+0000] {spark_submit.py:649} INFO - com.amazonaws#aws-java-sdk-bundle;1.12.262 from central in [default]
[2025-05-02T02:06:25.360+0000] {spark_submit.py:649} INFO - io.delta#delta-spark_2.12;3.3.0 from central in [default]
[2025-05-02T02:06:25.360+0000] {spark_submit.py:649} INFO - io.delta#delta-storage;3.3.0 from central in [default]
[2025-05-02T02:06:25.361+0000] {spark_submit.py:649} INFO - org.antlr#antlr4-runtime;4.9.3 from central in [default]
[2025-05-02T02:06:25.363+0000] {spark_submit.py:649} INFO - org.apache.hadoop#hadoop-aws;3.3.4 from central in [default]
[2025-05-02T02:06:25.364+0000] {spark_submit.py:649} INFO - org.wildfly.openssl#wildfly-openssl;1.0.7.Final from central in [default]
[2025-05-02T02:06:25.365+0000] {spark_submit.py:649} INFO - ---------------------------------------------------------------------
[2025-05-02T02:06:25.366+0000] {spark_submit.py:649} INFO - |                  |            modules            ||   artifacts   |
[2025-05-02T02:06:25.366+0000] {spark_submit.py:649} INFO - |       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2025-05-02T02:06:25.367+0000] {spark_submit.py:649} INFO - ---------------------------------------------------------------------
[2025-05-02T02:06:25.368+0000] {spark_submit.py:649} INFO - |      default     |   6   |   0   |   0   |   0   ||   6   |   0   |
[2025-05-02T02:06:25.368+0000] {spark_submit.py:649} INFO - ---------------------------------------------------------------------
[2025-05-02T02:06:25.369+0000] {spark_submit.py:649} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-4ac394d8-c752-48c7-ab53-b7934fa07484
[2025-05-02T02:06:25.370+0000] {spark_submit.py:649} INFO - confs: [default]
[2025-05-02T02:06:25.375+0000] {spark_submit.py:649} INFO - 0 artifacts copied, 6 already retrieved (0kB/9ms)
[2025-05-02T02:06:25.737+0000] {spark_submit.py:649} INFO - 25/05/02 02:06:25 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-05-02T02:06:27.625+0000] {spark_submit.py:649} INFO - Traceback (most recent call last):
[2025-05-02T02:06:27.626+0000] {spark_submit.py:649} INFO - File "/opt/etl/enriched_curated/creating_dim_allergies.py", line 199, in <module>
[2025-05-02T02:06:27.631+0000] {spark_submit.py:649} INFO - main()
[2025-05-02T02:06:27.632+0000] {spark_submit.py:649} INFO - File "/opt/etl/enriched_curated/creating_dim_allergies.py", line 8, in main
[2025-05-02T02:06:27.634+0000] {spark_submit.py:649} INFO - with open('../config.yaml') as config_file:
[2025-05-02T02:06:27.635+0000] {spark_submit.py:649} INFO - ^^^^^^^^^^^^^^^^^^^^^^
[2025-05-02T02:06:27.635+0000] {spark_submit.py:649} INFO - FileNotFoundError: [Errno 2] No such file or directory: '../config.yaml'
[2025-05-02T02:06:27.682+0000] {spark_submit.py:649} INFO - 25/05/02 02:06:27 INFO ShutdownHookManager: Shutdown hook called
[2025-05-02T02:06:27.683+0000] {spark_submit.py:649} INFO - 25/05/02 02:06:27 INFO ShutdownHookManager: Deleting directory /tmp/spark-ef3248f9-a7a7-42bc-820a-230be4eeb2d3
[2025-05-02T02:06:27.719+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 575, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master local --packages io.delta:delta-spark_2.12:3.3.0,org.apache.hadoop:hadoop-aws:3.3.4 --name arrow-spark --deploy-mode client /opt/etl/enriched_curated/creating_dim_allergies.py. Error code is: 1.
[2025-05-02T02:06:27.726+0000] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=medical_dag_v17, task_id=dim_group.creating_dim_allergies, run_id=manual__2025-05-02T02:06:20.068197+00:00, execution_date=20250502T020620, start_date=20250502T020622, end_date=20250502T020627
[2025-05-02T02:06:27.765+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-05-02T02:06:27.766+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 167 for task dim_group.creating_dim_allergies (Cannot execute: spark-submit --master local --packages io.delta:delta-spark_2.12:3.3.0,org.apache.hadoop:hadoop-aws:3.3.4 --name arrow-spark --deploy-mode client /opt/etl/enriched_curated/creating_dim_allergies.py. Error code is: 1.; 5527)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 575, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master local --packages io.delta:delta-spark_2.12:3.3.0,org.apache.hadoop:hadoop-aws:3.3.4 --name arrow-spark --deploy-mode client /opt/etl/enriched_curated/creating_dim_allergies.py. Error code is: 1.
[2025-05-02T02:06:27.794+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-05-02T02:06:27.805+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-05-02T02:06:27.807+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
